source/common/network/listener_impl.cc

```diff
@@ -43,85 +41,92 @@ bool ListenerImpl::rejectCxOverGlobalLimit() {
   return AcceptedSocketImpl::acceptedSocketCount() >= global_cx_limit;
 }
 
-void ListenerImpl::listenCallback(evconnlistener*, evutil_socket_t fd, sockaddr* remote_addr,
-                                  int remote_addr_len, void* arg) {
-  ListenerImpl* listener = static_cast<ListenerImpl*>(arg);
-
-  // Wrap raw socket fd in IoHandle.
-  IoHandlePtr io_handle = SocketInterfaceSingleton::get().socket(fd);
-
-  if (rejectCxOverGlobalLimit()) {
-    // The global connection limit has been reached.
-    io_handle->close();
-    listener->cb_.onReject();
-    return;
+void ListenerImpl::onSocketEvent(short flags) {
+  ASSERT(flags & (Event::FileReadyType::Read));
+
+  while (1) {
```

Comment:
It would be good to have a limit on the number of accept calls done per wakeup and warn if we had to do too many accepts in a row as it indicates that the proxy may have fallen behind for some reason.

Reply:
As far as I know, libevent is not doing anything more complicated than this. But we could definitely improve it going forward.

Reply:
Please add a TODO here

Reply:
Done!